_posts/2019-08-08-hacking-for-cruise-control.md

```diff
@@ -0,0 +1,303 @@
+---
+layout: post
+title:  "Hacking Strimzi for Cruise Control"
+date: 2019-08-29
+author: kyle_liberti
+---
+
+Cruise Control is a highly customizable software tool for simplifying the monitoring and balancing of workloads across a Kafka cluster.
+The flexibility of Cruise Control makes it easy to customize for different environments and use cases.
+This makes it possible to extend Cruise Control to work with Strimzi in an Kubernetes environment with a reasonable amount of effort.
+This post will walk through how to hook Cruise Control up to a Strimzi deployed Kafka cluster.
+<!--more-->
+
+# Prerequisite Outline
+
+There are a few considerations and tradeoffs to think about before jumping in.
+For starters, you will need a Kafka cluster.
+At the time of this post, Cruise Control officially supports Kafka version 2.0.0 but should still be compatible with later versions of Kafka.
+This blog post will stick to using Kafka version 2.3.0 but feel free to experiment with other Kafka versions.
+
+The core challenges of making Cruise Control work with Strimzi are: 
+
+- Gathering Kafka Metrics for Cruise Control consumption
+- Enabling communication between Cruise Control and Zookeeper
+- Deploying Cruise Control in Kubernetes
+
+We're looking at having deeper Cruise Control integration in Strimzi in the future, but this post is about how to get Cruise Control working with Strimzi today. For this reason I will address these challenges in as few steps as possible.
+
+# Gathering Metrics
+In order to estimate partition activity and resources, Cruise Control needs a steady source of metric data from Kafka.
+Fortunately, Kafka tracks all sorts of metrics and makes them all accessible through pluggable components known as 'metric reporters'. 
+Kafka metric reporters designate what, where, when, and how metrics are prepared for export.
+For Cruise Control, each Kafka broker is equipped with a metric reporter that will publish a subset of the broker's metrics to a special topic. 
+Cruise Control will then consume the metrics from this topic to learn about the load experienced by each broker in the cluster.
+
+Kafka maintains two core types of metrics:
```

Comment:
Although this is a good description of metric reporters themselves, CC is all about the broker, and so you only need to explain enough about metric reporter _as used by the broker_ for the reader to understand the context behind why they're configuring this thing. I think you nailed that in the previous paragraph and this and the following paragraphs in this section really don't add much that the reader needs to know. (I can see that your enthusiastic to share what you've learned about metric reports, which is fine, but I don't think this is the blog post to do that – if you digress too much the reader might stop reading).

"""